### DISCUSSION TITLE
Is there a version of Caffeine targeting Java 52 (Java SE 8)?

### DISCUSSION THREAD

**Discussion by dylanmccurry-msft** [ben-manes/caffeine#832] [Q&A] [answered]
Hi,

I am attempting to use Caffeine in AEM as a Cloud Service, but we're receiving this error:

```
  cannot access com.github.benmanes.caffeine.cache.Cache
  bad class file: /root/.m2/repository/com/github/ben-manes/caffeine/caffeine/3.1.2/caffeine-3.1.2.jar(com/github/benmanes/caffeine/cache/Cache.class)
    class file has wrong version 55.0, should be 52.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.
```

We're using maven for dependencies.  Is there a version of Caffeine available for Java 52?

**Comment by ben-manes** [comment, marked as answer]
Yes, `2.9.3` is the latest on the v2 line which targets Java 8. This is still supported, where the minimum JDK was increased in v3 to migrate from Unsafe onto VarHandles (Java 9) for CAS on volatile fields (as AtomicReference is pretty heavy if per entry). There is less release activity for 2.x as users migrate onto later JDKs, but bugs will be fixed (it may take a request to backport if the user impact isn't known).
